src/CoreBundle/private/library/classes/TCMSTableEditor/TCMSTableEditorEndPoint.class.php

```diff
@@ -1776,7 +1791,9 @@ protected function _WriteDataToDatabase(&$oFields, &$oData, $bDataFromDatabase =
         // we need this because we use a redirect later and would not see the error message, we need to change this later to a nice error_logging or something
         if ($bWasInserted) {
             $this->LoadDataFromDatabase();
-            if ($this->IsQueryLoggingAllowed()) {
+            /** @var  $migrationRecorderStateHandler \ChameleonSystem\DatabaseMigrationBundle\Bridge\Chameleon\Recorder\MigrationRecorderStateHandler*/
+            $migrationRecorderStateHandler = ServiceLocator::get('chameleon_system_database_migration.recorder.migration_recorder_state_handler');
+            if ($this->IsQueryLoggingAllowed() && $migrationRecorderStateHandler->isDatabaseLoggingActive() && count($setFields) >= 1) {
```

Comment:
Was this check (erroneously) missing?

$migrationRecorderStateHandler->isDatabaseLoggingActive()

Reply:
No, the check already existed but was only done after all the migration query data models where already prepared which is inefficient